hail/src/main/scala/is/hail/expr/ir/Simplify.scala

```diff
@@ -134,8 +134,10 @@ object Simplify {
     case IsNA(x) if isDefinitelyDefined(x) => False()
 
     case Let(n1, v, Ref(n2, _)) if n1 == n2 => v
+    case AggLet(n1, v, Ref(n2, _)) if n1 == n2 => v
```

Comment:
I think this is wrong.  n2 is referring to something in env, but AggLet is binding something in aggEnv.

Reply:
it should always be false?

Reply:
are the agg binding rules below wrong?

Reply:
No, there could be shadowing: `(Let a v (AggLet a w (Ref a))` is correct and the ref should refer to the (first) let.

Reply:
ok, removed these rules.